rskj-core/src/main/java/org/ethereum/util/ByteUtil.java

```diff
@@ -480,25 +480,11 @@ public static boolean isAllZeroes(byte[] array) {
         return true;
     }
 
-    public static Set<ByteArrayWrapper> difference(Set<ByteArrayWrapper> setA, Set<ByteArrayWrapper> setB){
+    public static <T> Set<T> difference(Set<T> setA, Set<T> setB){
 
-        Set<ByteArrayWrapper> result = new HashSet<>();
-
-        for (ByteArrayWrapper elementA : setA){
-            boolean found = false;
-            for (ByteArrayWrapper elementB : setB){
-
-                if (elementA.equals(elementB)){
-                    found = true;
-                    break;
-                }
-            }
-            if (!found) {
-                result.add(elementA);
-            }
-        }
-
-        return result;
+        return setA.stream()
+                .filter(elementA -> !setB.contains(elementA))
+                .collect(Collectors.toSet());
```

Comment:
Maybe this can be implemented using just `java.util.Collection` methods:
```java
Collection<T> temp = new HashSet<>(setA)
temp.removeAll(setB);
return temp;
```